packages/fxa-settings/src/components/Settings/Nav/index.tsx

```diff
@@ -21,10 +41,82 @@ export const Nav = () => {
     )}`;
 
-  const activeClasses = 'font-bold text-blue-500 rounded-sm bg-grey-50';
+  useEffect(() => {
+    let observer: IntersectionObserver;
+    if (
+      profileRef?.current &&
+      profileLinkRef.current &&
+      securityRef?.current &&
+      securityLinkRef.current &&
+      connectedServicesRef?.current &&
+      connectedServicesLinkRef.current &&
+      dataCollectionRef?.current &&
+      dataCollectionLinkRef.current
+    ) {
+      const options = {
+        threshold: 0.85,
+      };
+      observer = new IntersectionObserver((entries, observer) => {
+        entries.forEach((entry) => {
+          // Sections must have an ID beginning with "name-section" where "name" matches
+          // the nav link href. e.g. <a href="security" should have a corresponding
+          // section ID, id="security-section". They also must have a section ref.
+          const anchorId = entry.target.id.substring(
+            0,
+            entry.target.id.indexOf('-section')
+          );
+
+          let navLink: HTMLAnchorElement | null;
+          switch (anchorId) {
+            case 'profile':
+              navLink = profileLinkRef.current;
+              break;
+            case 'security':
+              navLink = securityLinkRef.current;
+              break;
+            case 'connected-services':
+              navLink = connectedServicesLinkRef.current;
+              break;
+            case 'linked-accounts':
+              navLink = linkedAccountsLinkRef.current;
+              break;
+            case 'data-collection':
+              navLink = dataCollectionLinkRef.current;
+              break;
+            default:
+              navLink = profileLinkRef.current;
+          }
+
+          if (entry.isIntersecting) {
+            if (!navLink?.classList.contains(navActiveClass)) {
+              navLink?.classList.add(navActiveClass);
+            }
+          } else if (navLink?.classList.contains(navActiveClass)) {
+            navLink.classList.remove(navActiveClass);
+          }
+        });
+      }, options);
+      observer.observe(profileRef.current);
+      observer.observe(securityRef.current);
+      observer.observe(connectedServicesRef.current);
+      observer.observe(dataCollectionRef.current);
+      if (linkedAccountsRef?.current) {
+        observer.observe(linkedAccountsRef.current);
+      }
+    }
+    return () => observer.disconnect();
+  }, [
+    profileRef,
+    securityRef,
+    connectedServicesRef,
+    dataCollectionRef,
+    linkedAccountsRef,
+  ]);
+
   return (
     <nav
-      className="font-header fixed bg-white w-full inset-0 mt-24 me-24 desktop:mt-11 desktop:static desktop:bg-transparent text-xl desktop:text-base"
+      // top-[7.69rem] allows the sticky nav header to align exactly with first section heading
+      className="font-header fixed bg-white w-full inset-0 mt-24 me-24 desktop:mt-0 desktop:sticky desktop:top-[7.69rem] desktop:bg-transparent text-xl desktop:text-base"
```

Comment:
Why the very specific and custom `desktop:top-[7.69rem]`?

Reply:
See the in-code comment above.

While it is true we'll have to tweak this if the content (e.g. height) in the sticky header changes, this was the least hacky way I could get this looking right, and it still looks good when +/-'ing the page as well since it's rems. Other solutions require height calculations in JS unfortunately. Happy to update it if there's a better solution though.